### DISCUSSION TITLE
Torus-EVM provider doesn't persist after login?

### DISCUSSION THREAD

**Discussion by mjafri118** [Web3Auth/Web3Auth#891] [Help]
Hi, 

I'm trying to make re-connecting a wallet seamless on refresh using web3auth in React, but I'm reprompted to login via a social login upon refresh. I think the root cause is something to do with "taurus-evm" as an adapter. Here's how I know this: 

I have the metamask and the torus-evm adapter set up: 

```export const torusWalletAdapter = new TorusWalletAdapter({
  adapterSettings: {
    buttonPosition: "bottom-right",
  },
  loginSettings: {
    verifier: "google",
  },
  // initParams: {
  //   buildEnv: "polygon",
  // },
  clientId: WEB3AUTH_CLIENTID,
});

export const metamaskAdapter = new MetamaskAdapter({
  clientId: WEB3AUTH_CLIENTID,
});

export const web3auth = new Web3Auth({
  clientId: WEB3AUTH_CLIENTID, // Get your Client ID from Web3Auth Dashboard
  chainConfig: {
    chainNamespace: "eip155",
    chainId: "0x89", // hex of 137, polygon mainnet
    rpcTarget: "https://rpc.ankr.com/polygon",
    // Avoid using public rpcTarget in production.
    // Use services like Infura, Quicknode etc
    displayName: "Polygon Mainnet",
    blockExplorer: "https://polygonscan.com",
    ticker: "MATIC",
    tickerName: "Matic",
  },
  enableLogging: true
});

// it will add/update  the metamask adapter in to web3auth class
web3auth.configureAdapter(metamaskAdapter);
web3auth.configureAdapter(torusWalletAdapter);
```

When I run the following:
``` 
await web3auth.init()
provider = await web3auth.connectTo("metamask")
```

After having logged in before, the provider is cached in some way and I'm not reprompted to connect via metamask. 

However, when I run the following: 
``` 
await web3auth.init()
provider = await web3auth.connectTo("torus-evm")
```

After having logged into Google before, the provider is NOT cached, and I have to log back into Google every single time. Clearly the session is still active-- going to app.tor.us between and after refreshes shows that the user is still logged into Torus via Google. It's only that my React app isn't able to pick it up. 

The Web3Modal's of the world have a `web3Modal.cachedProvider` to recover providers. Is there an equivalent? 

tl;dr for social logins via @web3auth/modal & torus-evm's Google login (or any other social login), how do I recover the web3 provider without having to reprompt the user to login again?

**Comment by mjafri118** [comment]
Additionally: it is very clear that the extra reprompting with Login with Google gets re run due to the `await web3auth.init()`. 

Refreshing the page with `connectTo('metamask')` performs great. Just need help on torus-evm.
